scalding-commons-macros/src/main/scala/com/twitter/scalding/commons/macros/impl/ordser/ScroogeOuterOrderedBuf.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2014 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.twitter.scalding.commons.macros.impl.ordser
+
+import scala.language.experimental.macros
+import scala.reflect.macros.Context
+
+import com.twitter.scalding._
+import java.nio.ByteBuffer
+import com.twitter.scalding.serialization.OrderedSerialization
+import com.twitter.bijection.macros.impl.IsCaseClassImpl
+import com.twitter.scrooge.{ ThriftUnion, ThriftStruct }
+import com.twitter.scalding.macros.impl.ordser._
+
+object ScroogeOuterOrderedBuf {
+  def dispatch(c: Context): PartialFunction[c.Type, TreeOrderedBuf[c.type]] = {
+    import c.universe._
+
+    val pf: PartialFunction[c.Type, TreeOrderedBuf[c.type]] = {
+      case tpe if tpe <:< typeOf[ThriftStruct] && !(tpe <:< typeOf[ThriftUnion]) => ScroogeOuterOrderedBuf(c)(tpe)
+    }
+    pf
+  }
+
+  def apply(c: Context)(outerType: c.Type): TreeOrderedBuf[c.type] = {
+    import c.universe._
+    def freshT(id: String) = newTermName(c.fresh(id))
+
+    val variableID = (outerType.typeSymbol.fullName.hashCode.toLong + Int.MaxValue.toLong).toString
```

Comment:
what's the intent of the Int.MaxValue.toLong here? This looks a bit dangerous. Can you comment on the intent. I guess you are trying to get a unique variable ID, but I can't tell if the intent is that for two types of the same name they get the same ID. If any case, I don't see the point of the MaxValue part.